packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1510,6 +1715,33 @@ export class AccountHandler {
     return createResponse();
   }
 
+  async getCredentialsStatus(request: AuthRequest) {
+    this.log.begin('Account.getCredentialsStatus', request);
+
+    const email = (request.payload as any).email;
+    await this.customs.check(request, email, 'getCredentialsStatus');
+    const { accountRecord } = await this.signinUtils.checkCustomsAndLoadAccount(
+      request,
+      email,
+      true
+    );
+
+    if (accountRecord.disabledAt) {
+      throw error.cannotLoginWithEmail();
+    }
+
+    if (accountRecord.verifierSetAt <= 0) {
+      throw error.cannotLoginWithEmail();
+    }
+
+    const response = {
+      currentVersion: accountRecord.clientSalt ? 'v2' :'v1',
```

Comment:
This is fine but if there's gonna be a v3 I suspect the implementer will curse our names.

Reply:
Are you objecting to the use of clientSalt to determine the version?

Reply:
The presence of a salt and the number of PBKDF2 iterations are two things.  'v2' is salt and 650000 rounds (at least that's how I'm interpreting it).  But in that ternary, 'v2' is just salt.  And presumably, if and when there's a v3, it will also have a salt.

Reply:
Eh, it's really just saying that since there's a salt we know it's a V2 client side key stretched password, which represents a breaking change to how passwords are encrypted that encompassed both iterations and a shift to not using the email in the salt anymore. 

Would changing this to something like   `currenVersion: /\/quickStretchV2:/.test(accountRecord.clientSalt || '') ? 'v2' : 'v1';` be better?

For what it's worth, there was back and forth on whether or not the salt alone should encode version info or not, or if it should also be explicitly declared in the database. A couple people weighed in on this, and I didn't have much of an opinion either way. Would you rather see the client side key stretch version in a database a column?

Reply:
No, like I said, it's fine.

I did think about what it would take to implement v3 while I looked at some of your changes.  But we can leave that as an exercise to whoever is going to work on it.  For all I know they won't like anything we come up with.